src/Hazelcast.Net/Clustering/Authenticator.cs

```diff
@@ -282,7 +282,7 @@ internal MemberGroups ParsePartitionMemberGroups(ClientAuthenticationCodec.Respo
         
         var isContainsCPGroupLeaderIds = response.KeyValuePairs.TryGetValue(ClusterCPGroups.CPGroupsJsonField, out var jsonMessage);
         
-        if (isContainsCPGroupLeaderIds && string.IsNullOrEmpty(jsonMessage))
+        if (!isContainsCPGroupLeaderIds || string.IsNullOrEmpty(jsonMessage))
```

Comment:
minor: if you read this out aloud then it's difficult to follow: "if not is contains CP group leader Ids ..." I'd suggest changing the name so the logic reads easier.

Reply:
It makes sense. It's always good to have a fresh pair of eyes 👀 Fixed at https://github.com/hazelcast/hazelcast-csharp-client/pull/907/commits/7a6a44c4fe4d14654f458efd48f31189cbde1e19